Wallet create insert improvements

joshuef opened the pull request:
<!--
#### Thank you for contributing!

Please reference the issue(s) which this pull request addresses using [keywords](https://help.github.com/articles/closing-issues-using-keywords/) such as:

```
Resolves #452
Fixes #363
Closes #408
```

---

Provide QA team and reviewer steps to test the resolution.
For example:

```
QA:
Easiest way to test this PR would be to:
- Run safe-auth cli client in a different terminal
- login
- safe wallet insert <problem link>
- Expect to see an authentication error output

```

---

Commit messages should conform to the format:

```
<type>(<scope>): <description>

[optional body]

```

For example:

```
fix(auth): proper values returned on auth_decode_ipc_msg errors

  - Test case for authenticator error -208 IncompatibleMockStatus
  - Test case for authenticator error -103 when decoding share MData
    request for non-existent MData

```

Commit `type` can be one of:
**feat**: New feature
**fix**: Bug fix
**docs**: Documentation only changes
**style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
**refactor**: A code change that neither fixes a bug nor adds a feature
**perf**: A code change that improves performance
**test**: Adding missing tests
**chore**: Changes to the build process or auxiliary tools and libraries such as documentation generation
**revert**: Reverting a feature, fix, or commit which introduces a regression or new bug

Commit `scope` is open to any succinct term which indicates the effected feature or component.

See [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)

---
Write your description below this line: -->

bochaco commented on src/subcommands/wallet.rs:
```diff
@@ -13,27 +13,22 @@ use unwrap::unwrap;
         /// The name to give this spendable balance
         #[structopt(long = "name")]
         name: Option<String>,
-        /// Create a Key, allocate test-coins onto it, and add it to the Wallet
```
On top of my previous comment about removing the abulity to aut-create new Key hen inserting I do agree with removing this options though, so I suggest we still allow to auto-create a Key when inserting but just empty with no option for preloading it,or at least not for now to keep the command as simple as possible

joshuef commented on README.md:
```diff
@@ -228,9 +255,9 @@ OPTIONS:
+    <key>        An existing `Key`'s safe://xor-url.
     <source>     The source Wallet for funds
     <target>     The target Wallet to insert the spendable balance
-    <key>        An existing `Key`'s safe://xor-url. If this is not supplied, a new `Key` will be automatically generated and inserted
```
Hmm, yeh maybs (although I think that use case isn't necessarily required for SAFE). but i'd not do it automagically. maybe with a `--new` arg?

bochaco replied:
yes a `--new` or `--generate` could be better yes. Also, I was thinking this PR would be solveing issue #81 but I don't think you are adding the `--default` option, shall we? either way we should link this PR to that issue

bochaco replied:
I'm just thinking that for whatever reason you wanna add new receiving Key to a Wallet it's a nice feature if that can be done in one operation rather than two (key create + wallet insert)

joshuef replied:
It makes it `default` by automagic. So yeh, solves #81

joshuef replied:
Hmmm, thinking on this more, the generation should be possible with some simple piping. I'm not sure we need to cover _everything_ in each command. 

"Do one thing and do it well..."

joshuef replied:
@hunterlester thoughts?

hunterlester replied:
Keeping `Keys` and `Wallet` commands separate, and doing something like `safe keys create | safe wallet insert --target <xor-url>` or `safe keys create | safe wallet create`??
Vs `safe wallet --new` <-- created with generated key
The former would be nicer for us :-) Easier to test.
Latter is much nicer for user, but then again CLI users may already be used to writing scripts to manage the verbosity of former.

joshuef replied:
Right now, with this PR we have `safe wallet create` which creates a new wallet AND generates spendable balance.

We also have `safe wallet insert <key>` to allow adding a key. But there's (with this PR), _no helper for inserting a newly generated key. (Ie, before this PR you could only `safe wallet insert <target wallet xor>`, and it would generate a new key on the wallet.)

I don't think we need this as a) It's quite confusing how to use `insert` then. And b) The same result could be achieved with `safe keys create | safe wallet insert <target wallet>`, with the former piping the keys xor to the wallet eg.

hunterlester replied:
Yeah, I like the simpler commands and opting for piping outputs to other inputs.

joshuef commented on README.md:
````diff
@@ -197,12 +197,38 @@ SUBCOMMANDS:
+For example, we can create a new `Wallet` with a new spendable balance by simply running:
+
+```shell
+$ safe_cli wallet create --pretty --source <source wallet to pay for PUTs>
````
This is done.

joshuef commented:
Updated the PR and fixed the failing tests there. Only thing left is to consider auto gen for insert or be happy w/ piping `keys create | wallet insert` style commands.
